api-controllers/ServiceFabrikApiController.js

```diff
@@ -361,32 +385,37 @@ class ServiceFabrikApiController extends FabrikBaseController {
     const backupStartedAt = new Date();
     return eventmesh
       .apiServerClient.getLastOperation({
-        resourceId: req.params.instance_id,
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
         operationName: CONST.OPERATION_TYPE.BACKUP,
         operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+        resourceId: req.params.instance_id
       })
       .then(backupGuid => {
         return eventmesh
-          .apiServerClient.getOperationState({
-            operationName: CONST.OPERATION_TYPE.BACKUP,
-            operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-            operationId: backupGuid,
+          .apiServerClient.getResource({
+            resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+            resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+            resourceId: backupGuid
           })
-          .then(state => {
+          .then(response => {
+            const state = response.status.state;
             // abort only if the state is in progress
             if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
-              return eventmesh.apiServerClient.updateOperationState({
-                operationName: CONST.OPERATION_TYPE.BACKUP,
-                operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-                operationId: backupGuid,
-                stateValue: CONST.OPERATION.ABORT
+              return eventmesh.apiServerClient.updateResource({
```

Comment:
What updateResource does?
It patches defined fields in the resource for ex:
`status.state` can be overwritten or `status.response` can be overwritten. But what if we just want to add few more properties in status.response which is a stringified. `patchResponse` is used for that.

We can have a wrapper method for `patchStatus` or `patchOptions` but it can anyway achieved with `updateResource`  by passing only `status` let's say in `opts`

Reply:
> We can have a wrapper method for patchStatus or patchOptions but it can anyway achieved with updateResource by passing only status let's say in opts

Above is exactly my Q. Why have we defined these methods `patchStatus` and `patchResponse` when same could be achieved via updateResource? Somehow i feel this a bit confusing as its not consistent. 

Also, since these resource objects are so central to the understanding, can we have each resource type  documented in wiki (sample formatted JSON dump would also do)?

Reply:
Now we have one method named `patchResource` which handles the appending of `spec.options` as well as `status.response`